xbmc/pvr/channels/PVRChannelGroups.cpp

```diff
@@ -240,7 +245,11 @@ bool CPVRChannelGroups::UpdateGroupsEntries(const CPVRChannelGroups &groups)
 
     // add it if not
     if (!existingGroup)
-      m_groups.push_back(CPVRChannelGroupPtr(new CPVRChannelGroup(m_bRadio, -1, (*it)->GroupName())));
+    {
+      existingGroup = *(m_groups.insert(m_groups.end(),CPVRChannelGroupPtr(new CPVRChannelGroup())));
```

Comment:
I use it always this way (insert) because push_back doesn't return the inserted iterator.
But I'm also fine with your convention. I'll change it tomorrow